src/connection.rs

```diff
@@ -56,6 +58,8 @@ pub struct Connection {
     close_tx: UnboundedSender<SubstreamId>,
     close_rx: UnboundedReceiver<SubstreamId>,
 
+    pub(crate) message_nonce: Arc<Mutex<u64>>,
```

Comment:
This works, but this could be replaced with `Arc<AtomicU64>`.

Reply:
Reading and storing the value will have to specify [Ordering::SeqCst](https://doc.rust-lang.org/std/sync/atomic/enum.Ordering.html#variant.SeqCst) to ensure sequentially consistent operations.

Reply:
updated